design/apidsl/type.go

```diff
@@ -59,6 +63,7 @@ func Type(name string, dsl func()) *design.UserTypeDefinition {
 	return t
 }
 
+// ArrayOf can be used in: (it is a top level DSL)
```

Comment:
`ArrayOf` can be used wherever a type can be used, e.g. `Attribute("foo", ArrayOf(String))`